crates/common/trie/node.rs

```diff
@@ -83,6 +113,12 @@ impl NodeRef {
             NodeRef::Hash(hash) => *hash,
         }
     }
+
+    pub fn clear_hash(&mut self) {
```

Comment:
We should document what this does and why it's needed. If we can change this to be private, even better.